src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -244,16 +258,14 @@ internal sealed class SynthesizedComplexParameterSymbol : SynthesizedParameterSy
             RefKind refKind,
             string name,
             ImmutableArray<CustomModifier> refCustomModifiers,
-            ImmutableArray<CSharpAttributeData> attributes,
-            bool hasEnumeratorCancellationAttribute)
+            SourceComplexParameterSymbol? baseParameterForAttributes)
             : base(container, type, ordinal, refKind, name)
         {
             Debug.Assert(!refCustomModifiers.IsDefault);
-            Debug.Assert(!attributes.IsDefault);
+            Debug.Assert(!refCustomModifiers.IsEmpty || baseParameterForAttributes is object);
```

Comment:
>!refCustomModifiers.IsEmpty || baseParameterForAttributes is object [](start = 25, length = 67)

It is not obvious why is this the right thing to assert

Reply:
If this condition is false, it means there was no good reason to create this as a SynthesizedComplexParameterSymbol instead of a SynthesizedParameterSymbol.

---
In reply to: [373192152](https://github.com/dotnet/roslyn/pull/41299#discussion_r373192152) [](ancestors = 373192152)